pyro/poutine/replay_messenger.py

```diff
@@ -57,19 +32,23 @@ def _pyro_sample(self, msg):
         reverts to default Messenger._pyro_sample behavior with no additional side effects.
         """
         name = msg["name"]
-        # case 1: dict, positive: sample from guide
-        if name in self.sites:
+        if self.trace is not None and name in self.trace:
+            guide_msg = self.trace.nodes[name]
             if msg["is_observed"]:
-                raise RuntimeError("site {} is observed and should not be overwritten".format(name))
-            g_name = self.sites[name]
-            if g_name not in self.trace:
-                raise RuntimeError("{} in sites but {} not in trace".format(name, g_name))
-            if self.trace.nodes[g_name]["type"] != "sample" or \
-                    self.trace.nodes[g_name]["is_observed"]:
-                raise RuntimeError("site {} must be sample in trace".format(g_name))
+                return None
+            if guide_msg["type"] != "sample" or \
+                    guide_msg["is_observed"]:
+                raise RuntimeError("site {} must be sample in trace".format(name))
             msg["done"] = True
-            msg["value"] = self.trace.nodes[g_name]["value"]
+            msg["value"] = guide_msg["value"]
+            msg["infer"] = guide_msg["infer"]
```

Comment:
Do I understand correctly that we never needed this before since we only replayed models against guides, but now we're replaying guides against guides as well? Or are you merely cleaning up?

Reply:
Just cleanup, the sites logic never got used